crates/ra_ide/src/lib.rs

```diff
@@ -494,6 +494,10 @@ impl Analysis {
     }
 }
 
+pub fn check_structural_search_replace_query(query: &str) -> Option<SsrError> {
```

Comment:
```suggestion
pub fn check_ssr_query(query: &str) -> Option<SsrError> {
```

Reply:
should I rename `structural_search_replace(...)` it's about ten lines above as well? I don't mind just want to keep it consistent.

Reply:
Um.. I think making it solely consists with `handle_check_ssr_query` is fine.

Reply:
Another option is to remove this function altogether and just using `query.parse::<SsrQuery>`  at the call site.